gdnative-core/src/core_types/vector2.rs

```diff
@@ -89,6 +91,15 @@ impl Vector2Godot for Vector2 {
         Vector2::new(self.y, -self.x)
     }
 
+    #[inline]
+    fn normalized(self) -> Self {
+        if self.length() > 1.0 {
```

Comment:
Should this check really be here? If the length of the vector is less than one it also isn't a unit vector. It should always divide by the length of the vector without doing checks on the current length

Reply:
im new to godot and rust but from my tinkering in godot i got this results, when i use:
`if input_vector.length() > 1.0 {`
`input_vector = input_vector / input_vector.length(); }`

everything is fine and works, but if i try without `if`:
`input_vector = input_vector / input_vector.length();`
result vector is incorrect: `(NaN, NaN)`
from my observation problem with NaN is when vector has value of (0.0, 0.0) - division by zero.

Reply:
The godot version of vector2 just uses `return Math::sqrt(x * x + y * y);` which is the correct length calculation. What I've just noticed is that there should be a check but with `self.length() != 0.0` or `self.length_squared() != 0.0` to avoid a division by zero but otherwise it should always do the division. The C++ version uses the squared length and then calculates the square root of that after the check for better performance. What input vector did you use to get a `(NaN, NaN)` result?

Reply:
this is my code:
`let mut input_vector = Vector2::zero();`
`input_vector.x = Input::get_action_strength(input, "ui_right") as f32 - Input::get_action_strength(input, "ui_left") as f32;`
`input_vector.y = Input::get_action_strength(input, "ui_down") as f32 - Input::get_action_strength(input, "ui_up") as f32;`
its inside `_physics_process` so its continuously updated, and when no arrows are pressed then you have:
`input_vector =  (0.0, 0.0)`

Reply:
The current type should already have a `normalize()` method that handles any non `(0, 0)` cases properly and to prevent a panic you can also use `try_normalize()`: https://docs.rs/euclid/0.22.0/euclid/struct.Vector2D.html#method.try_normalize

Reply:
yep, i tried `normalize()` and get `NaN`:
`input_vector = input_vector.normalize();`
and
`input_vector = Vector2::normalize(input_vector);`

Reply:
I think that's because your vector is `(0, 0)`, normalising is not defined for vectors of length 0. You can use `try_normalize()` which will give you an `Option` instead so you can check if you actually *can* normalise the vector.

Reply:
You are right, problem is that my vector can get `(0, 0)` value, so of course i can manually write check every time before i use normalize, but i think there should be normalize function that would automatically check vector value, maybe not my hacky solution, but instead port godot solution, for this function. I think `try_normalize()` is not solution for this problem, you still need to manually write check result because `returns None in the case that the length of the vector is zero.` so if it returns `None` so you need to manually write code that's sets vector back to `(0, 0)`.

Reply:
You can always add any convenience extension methods with any arbitrary properties you want in your own crate.

Reply:
ok, so i made my internal function:

`fn normalized(vector_to_normalize: Vector2) -> Vector2 {`
    `let option = Vector2::try_normalize(vector_to_normalize);`
    `match option {`
        `None => Vector2::zero(),`
        `Some(vector2) => vector2,}}`